packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -339,6 +350,18 @@ library ConnextLogic {
     emit MaxRoutersPerTransferUpdated(_newMax, msg.sender);
   }
 
+  /**
+   * @notice Used to set the sponsor vault
+   * @param _sponsorVault The new sponsor vault
+   * @param _currentSponsorVault The current sponsor vault
+   */
+  function setSponsorVault(address _sponsorVault, address _currentSponsorVault) external {
```

Comment:
we need to require that this request comes from the `_currentSponsorVault` or something... otherwise, it could be a griefing vector to overwrite sponsor's vaults freely

Reply:
ah wait, misinterpreted the use case entirely. The `setSponsorVault` call comes from the `ConnextHandler`, which requires the owner. My b.